src/main/java/stack/MatchingBrackets.java

```diff
@@ -22,21 +22,20 @@ public int balanced(String string) {
                     if (topElement == matchingbracket) {
                         stack.pop();
                     } else {
-                        return i + 1 ;
+                        return i + 1;
                     }
-                    index = (string.indexOf(topElement));
                 }
             }
         }
-
         if (stack.empty()) {
             return -1;
         } else {
-           return index;
+            i = (string.lastIndexOf(stack.peek()) + 1);
```

Comment:
This will still not work. Think about example `{{{}}`. What should your code returns and what should it return?

Don't always wait for me to give counter examples, whenever you make a change test it on a number of examples before you commit it. Did you test this change on multiple examples?

Reply:
Here is a hint:
Because of duplicate characters, you cannot use `indexof` or `lastIndexof` as they will either give you first or last index of the character while answer could be first, last or any of the middle index depending on example. You will have to remember the index(es) in some data structure as you are processing the string.